### ISSUE TITLE
segfault

### ISSUE THREAD

**Issue by sdahlbac** [cyrilgdn/terraform-provider-postgresql#225]
Hi there,

Thank you for opening an issue. Please provide the following information:

### Terraform Version
Run `terraform -v` to show the version. If you are not running the latest version of Terraform, please upgrade because your issue may have already been fixed.

terraform -v
Terraform v1.2.4
on darwin_arm64
+ provider registry.terraform.io/aztfmod/azurecaf v2.0.0-preview-3
+ provider registry.terraform.io/cyrilgdn/postgresql v1.16.0
+ provider registry.terraform.io/hashicorp/azurerm v3.13.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.12.1

### Terraform Configuration Files
```hcl
provider "postgresql" {
  host     = data.terraform_remote_state.base.outputs.db-fqdn
  username = "postgres"
  password = data.terraform_remote_state.base.outputs.db-admin-username
  port     = 5432
  sslmode  = "require"

  expected_version = "13.0.0"
}

resource "postgresql_database" "fabric" {
  name     = "fabric_${terraform.workspace}"
  owner    = data.terraform_remote_state.base.outputs.db-admin-username
  encoding = "UTF8"
}
```
### Panic Output
If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`.

Stack trace from the terraform-provider-postgresql_v1.16.0 plugin:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x1049f5cc4]

goroutine 83 [running]:
database/sql.(*Tx).grabConn(0x0, {0x104eeed70, 0x1400003c178})
	database/sql/sql.go:2190 +0x44
database/sql.(*Tx).ExecContext(0x0, {0x104eeed70, 0x1400003c178}, {0x104bcfafe, 0x19}, {0x0, 0x0, 0x0})
	database/sql/sql.go:2446 +0x38
database/sql.(*Tx).Exec(...)
	database/sql/sql.go:2459
github.com/terraform-providers/terraform-provider-postgresql/postgresql.pgLockRole(0x0, {0x140006982a8, 0x8})
	github.com/terraform-providers/terraform-provider-postgresql/postgresql/helpers.go:463 +0x60
github.com/terraform-providers/terraform-provider-postgresql/postgresql.createDatabase(0x140006a3500, 0x1400066d580)
	github.com/terraform-providers/terraform-provider-postgresql/postgresql/resource_postgresql_database.go:128 +0x110
github.com/terraform-providers/terraform-provider-postgresql/postgresql.resourcePostgreSQLDatabaseCreate(0x140006a3500, 0x1400066d580)
	github.com/terraform-providers/terraform-provider-postgresql/postgresql/resource_postgresql_database.go:110 +0x30
github.com/terraform-providers/terraform-provider-postgresql/postgresql.PGResourceFunc.func1(0x1400066d580, {0x104d9ece0, 0x140002e2d80})
	github.com/terraform-providers/terraform-provider-postgresql/postgresql/helpers.go:22 +0x84
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x1400021a9a0, {0x104eeede0, 0x14000653920}, 0x1400066d580, {0x104d9ece0, 0x140002e2d80})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.15.0/helper/schema/resource.go:695 +0x170
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x1400021a9a0, {0x104eeede0, 0x14000653920}, 0x14000674a90, 0x1400066d400, {0x104d9ece0, 0x140002e2d80})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.15.0/helper/schema/resource.go:837 +0xc14
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x1400045acd8, {0x104eeed38, 0x140006c0d00}, 0x14000654780)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.15.0/helper/schema/grpc_provider.go:1021 +0xedc
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x1400043eaa0, {0x104eeede0, 0x14000653170}, 0x140006701c0)
	github.com/hashicorp/terraform-plugin-go@v0.9.0/tfprotov5/tf5server/server.go:812 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x104e921a0, 0x1400043eaa0}, {0x104eeede0, 0x14000653170}, 0x140006a2ba0, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.9.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x14000210e00, {0x104efe498, 0x140005081a0}, 0x1400065cc60, 0x140004f01b0, 0x1054c0da0, 0x0)
	google.golang.org/grpc@v1.45.0/server.go:1282 +0xc5c
google.golang.org/grpc.(*Server).handleStream(0x14000210e00, {0x104efe498, 0x140005081a0}, 0x1400065cc60, 0x0)
	google.golang.org/grpc@v1.45.0/server.go:1619 +0xa34
google.golang.org/grpc.(*Server).serveStreams.func1.2(0x14000518520, 0x14000210e00, {0x104efe498, 0x140005081a0}, 0x1400065cc60)
	google.golang.org/grpc@v1.45.0/server.go:921 +0x94
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.45.0/server.go:919 +0x1f0

Error: The terraform-provider-postgresql_v1.16.0 plugin crashed!
```

### Expected Behavior
What should have happened?

no crash

### Actual Behavior
What actually happened?

crash

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

**Comment by sdahlbac** [comment]
This is probably the same as #183 basically bad credentials. Still, bad credentials should not lead to a segfault

**Comment by dCode0** [comment]
Do we have an update on this?

│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.

Error: The terraform-provider-postgresql_v1.16.0 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

**Comment by cyrilgdn** [comment]
Hi @sdahlbac ,

Did you confirmed it was bad credentials?

I've just tested and for bad credentials I have this error:
```
Error: error detecting capabilities: error PostgreSQL version: pq: password authentication failed for user "postgres"
```
which is clear enough I think.

**Comment by cyrilgdn** [comment]
Closing as duplicate of #183 actually.

**Comment by cyrilgdn** [comment]
Fixed in v1.17.1
